hca2scea-backend/fetch_fastq_path.py

```diff
@@ -0,0 +1,206 @@
+import argparse
+import os
+import requests as rq
+from xml.etree import ElementTree
+import multiprocessing
+from contextlib import contextmanager
+import pandas as pd
+
+@contextmanager
+def poolcontext(*args, **kwargs):
+    pool = multiprocessing.Pool(*args, **kwargs)
+    yield pool
+    pool.terminate()
+
+def parse_xml(xml_content):
+    for experiment_package in xml_content.findall('EXPERIMENT_PACKAGE'):
+        yield experiment_package
+
+def filter_paths(sdrf, paths):
+    runs = list(sdrf['Comment[ENA_RUN]'])
```

Comment:
cleanup tip:
This a classic TMI method (Too Much Information)
There is too much detail here.
I would extract bits of it to properly named functions.
For example, the read1 bits, the read2 bits, processing a run.
Would you like to do a few examples together?